### PR TITLE
ARROW-2778: [JS] Add Utf8Vector.from

### PR DESCRIPTION
**Pull request by TheNeuralBit**

### CODE REVIEW

**Comment by trxcllnt** [comment]
@TheNeuralBit looks good to me -- do you think it's worth checking for nulls and building the bitmap here too, or should that go in a more full featured "builder" api of some sort?

File: js/src/vector.ts [hunk 1]
```diff
@@ -348,7 +350,15 @@ export class FixedSizeBinaryVector extends FlatVector<FixedSizeBinary> {
     }
 }
 
+const utf8Encoder = new TextEncoder('utf-8');
+
 export class Utf8Vector extends ListVectorBase<Utf8> {
+    static from(values: string[]) {
```
Comment (trxcllnt) [inline js/src/vector.ts]: We have a bunch of code internally at Graphistry that we could contribute that does essentially this for a number of the types. For example, going from a random JS Array to a Dictionary encoded binary or UTF-8 vector, with null handling. We wrote it mostly to convert between existing protobuf data + random JS arrays, so it allows for customizing the encoding a bit with selectors (e.g. you can pass in a function to indicate the string `"\0"` should be treated as a null). Ex: https://gist.github.com/trxcllnt/2b4216fc95288c2e0ff92ecf826376e0#file-arrow-encode-ts-L34

Comment (trxcllnt) [reply js/src/vector.ts]: @TheNeuralBit I can push that stuff up to this PR if you like

Comment (TheNeuralBit) [reply js/src/vector.ts]: oh yeah that would be awesome!

**Comment by wesm** [comment]
@trxcllnt @TheNeuralBit is this obsolete with the refactor branch?

**Comment by trxcllnt** [comment]
@wesm yes, this feature was included in the refactor branch

**Comment by wesm** [comment]
:+1:

**Pull request closed without merging**